presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayAbstractMinMaxFunction.java

```diff
@@ -0,0 +1,187 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.metadata.FunctionInfo;
+import com.facebook.presto.metadata.FunctionRegistry;
+import com.facebook.presto.metadata.OperatorType;
+import com.facebook.presto.metadata.ParametricScalar;
+import com.facebook.presto.metadata.Signature;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.type.StandardTypes;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.TypeManager;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.slice.Slice;
+
+import java.lang.invoke.MethodHandle;
+import java.util.Map;
+
+import static com.facebook.presto.metadata.Signature.comparableTypeParameter;
+import static com.facebook.presto.spi.StandardErrorCode.INTERNAL_ERROR;
+import static com.facebook.presto.type.TypeUtils.parameterizedTypeName;
+import static com.facebook.presto.type.TypeUtils.readStructuralBlock;
+import static com.facebook.presto.util.Reflection.methodHandle;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public abstract class ArrayAbstractMinMaxFunction
+        extends ParametricScalar
+{
+    private final OperatorType operatorType;
+    private final String functionName;
+    private final Signature signature;
+
+    private static final Map<Class<?>, MethodHandle> METHOD_HANDLES = ImmutableMap.<Class<?>, MethodHandle>builder()
+            .put(boolean.class, methodHandle(ArrayAbstractMinMaxFunction.class, "booleanArrayMinMax", MethodHandle.class, Type.class, Slice.class))
+            .put(long.class, methodHandle(ArrayAbstractMinMaxFunction.class, "longArrayMinMax", MethodHandle.class, Type.class, Slice.class))
+            .put(double.class, methodHandle(ArrayAbstractMinMaxFunction.class, "doubleArrayMinMax", MethodHandle.class, Type.class, Slice.class))
+            .put(Slice.class, methodHandle(ArrayAbstractMinMaxFunction.class, "sliceArrayMinMax", MethodHandle.class, Type.class, Slice.class))
+            .put(void.class, methodHandle(ArrayAbstractMinMaxFunction.class, "arrayWithUnknownType", MethodHandle.class, Type.class, Slice.class))
+            .build();
+
+    public ArrayAbstractMinMaxFunction(OperatorType operatorType, String functionName)
+    {
+        this.operatorType = operatorType;
+        this.functionName = functionName;
+        this.signature = new Signature(functionName, ImmutableList.of(comparableTypeParameter("E")), "E", ImmutableList.of("array<E>"), false, false);
```

Comment:
Did you mean `comparable` means `=`, eg `equalTo`? `orderable` is `compareTo`.
So what we need here is `orderableType`
It's a little confusing to me that `comparableType` might not support `compareTo`